### PR TITLE
MGMT-2703 - optionally wait for CVO to become available after install…

### PR DESCRIPTION
**Pull request by yevgeny-shnaidman**
…ation

### CODE REVIEW

File: discovery-infra/utils.py [hunk 1]
```diff
@@ -525,3 +549,15 @@ def set_hosts_roles_based_on_requested_name(client, cluster_id):
     client.set_hosts_roles(cluster_id=cluster_id, hosts_with_roles=hosts_with_roles)
 
 
+def config_etc_hosts(cluster):
```
Comment (tsorya) [inline discovery-infra/utils.py:552]: u don't need it. With latest commit from @YuviGold  u have dns already

File: discovery-infra/start_discovery.py [hunk 2]
```diff
@@ -347,6 +347,10 @@ def nodes_flow(client, cluster_name, cluster, image_path):
             )
             # Validate DNS domains resolvability
             validate_dns(client, cluster.id)
+            if args.wait_for_cvo:
+                cluster_info = client.cluster_get(cluster.id)
+                utils.config_etc_hosts(cluster_info)
```
Comment (romfreiman) [inline discovery-infra/start_discovery.py:352]: why isnt it the default? Shouldnt we always do some verification on the installed OCP?

File: discovery-infra/utils.py [hunk 3]
```diff
@@ -102,6 +102,29 @@ def get_libvirt_nodes_mac_role_ip_and_name(network_name):
...
+        sleep_seconds=20,
+        waiting_for="CVO to become available",
+    )
+
+
+def is_cvo_available():
+    try:
+        res = subprocess.check_output("kubectl --kubeconfig=build/kubeconfig get clusterversion -o json", shell=True)
+        conditions = json.loads(res)['items'][0]['status']['conditions']
+        for condition in conditions:
```
Comment (romfreiman) [inline discovery-infra/utils.py:118]: maybe we should print the output here? Otherwise it will be hard to debug, no?

**Comment by romfreiman** [comment]
You can adddress my comments once it's merged.

@yevgeny-shnaidman @YuviGold how will prow execute this?

File: discovery-infra/start_discovery.py (hunk 2 above)
Comment (yevgeny-shnaidman) [reply discovery-infra/start_discovery.py:352]: i think that right now most of the people want to check the usual install. I have added a new target for those that want to wait for CVO and debug.That way we don't interfere with CI and PRs

File: discovery-infra/utils.py (hunk 3 above)
Comment (yevgeny-shnaidman) [reply discovery-infra/utils.py:118]: the main debugging will be done by the must-gather.  right now, even when we have issues with etcd, what we will see in the conditions is: multiple errors with api server endpoint oath. It does not tell much

Comment (tsorya) [reply discovery-infra/utils.py:118]: u can look on failure status, u will all those errors

File: discovery-infra/start_discovery.py (hunk 2 above)
Comment (romfreiman) [reply discovery-infra/start_discovery.py:352]: sorry if I was not clear - I specifically addressed  utils.config_etc_hosts(cluster_info), not the whole function

Comment (yevgeny-shnaidman) [reply discovery-infra/start_discovery.py:352]: aaa-a-a. 
1) right now we need it only for CVO
2) this whole issue should be solved by @YuviGold commit of set_dns. unfortunately, on some servers ( for example HW Lab) this cause issues with resolv.conf, which causes the whole flow to fail (@tsorya wa trying to look into it). I prefer to push it with changes to etc/hosts and not to wait till the set_dns is foixed. Once the fix is in place, i will simply remove etc/hosts configurations

File: discovery-infra/utils.py (hunk 3 above)
Comment (tsorya) [reply discovery-infra/utils.py:118]: i think we miss logs here

**Pull request merged**